application-task-ui/src/main/resources/TaskManager/GanttMacro/JSONTaskSource.xml

```diff
@@ -0,0 +1,195 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.4" reference="TaskManager.GanttMacro.JSONTaskSource" locale="">
+  <web>TaskManager.GanttMacro</web>
+  <name>JSONTaskSource</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>xwiki:TaskManager.GanttMacro</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title/>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity wiki="false"}}
+#macro (putOrDefault $map $propertyName $value $default1)
+  #if ($value)
+    #set ($discard = $map.put($propertyName, $value))
+  #else
+    #set ($discard = $map.put($propertyName, $default1))
+  #end
+#end
+
+#macro (csvToArray $csvString $return)
+  ## When a project name contains a comma, it can be escaped with a backslash.
```

Comment:
TODO for me: to investigate what is the best practice for fields with comma separated values

Reply:
Update: I [implemented](https://github.com/KebabRonin/application-task/tree/parameter-list-escapes-beta) the backslash escapes, but even though it works for now, backslashes inside names can still lead to some unforeseen consequences:
* REST APIs not working (tested by trying to update a task status in the `kanbanboard` macro)
* The `$xwiki.getURL()` method returns the formatted names, so links to spaces which conflict with the naming strategy lead to non-existent pages
  * Since the default behavior is to not check page titles on save, creating a new page through a non-existent url will succeed, even if the name contains (url escaped) backslashes

Reply:
Todo: check if `$wikimacro.parameters` can be used https://www.xwiki.org/xwiki/bin/view/Documentation/DevGuide/Tutorials/WritingMacros/WikiMacroTutorial/ , or it is not working because it cannot deserialize the String value